packages/ibm-products/src/components/ConditionBuilder/ConditionGroupBuilder/ConditionGroupBuilder.js

```diff
@@ -45,12 +46,46 @@ const ConditionGroupBuilder = ({
       variant == 'tree'
         ? handleFocusOnCloseTree(evt)
         : handleFocusOnClose(evt, conditionIndex);
-      onChange({
-        ...group,
-        conditions: group.conditions.filter(
-          (condition) => conditionId !== condition.id
-        ),
-      });
+
+      if (group.conditions[1].conditions) {
+        //when we remove every plain conditions of a group without deleting  the subgroup, we need to restructure the group
+        //the inner group become outer group and same level subgroups become plain conditions
+
+        let newGroup = [],
```

Comment:
since `newGroup` isn't actually used till line 73 i would just move it's instantiation to there. `const newGroup = [ ... ]`

Reply:
resolved